### ISSUE TITLE
Clean up the duplicated (and problematic) "generate-go-crd-clients"

### ISSUE THREAD

**Issue by yuwenma** [GoogleCloudPlatform/k8s-config-connector#3037]
### Checklist

- [X] I did not find a related open enhancement request.
- [X] I understand that enhancement requests filed in the GitHub repository are by default low priority.
- [X] If this request is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Describe the feature or resource

*This issue is for tracking purpose*

This originated from triaging a bug that block [promoting a KMS Alpha resource to Beta](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3012) (direct), where the KMS services have both TF and direct resources in both Alpha and Beta versions, and the autogenerated client cannot import the autogenerated API (alpha), when promoting alpha to beta and keep the alpha CRD. It turns out that the bug comes from `generate-go-crd-clients` where it cannot support auto-generating APIs with multiple CRD versions. Instead, it deletes the autogenerated alpha API.

The real problem is that the  `generate-go-crd-clients`  generates the api and go client lib using go template, rather than relying on the sig-k8s its own solution (controller-gen). 

As we migrating to the [new "direct" approach](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/docs/develop-resources), we are using the controller-gen to build the api client and can now support multiple CRD versions. However, the  `generate-go-crd-clients`  is still there in the presubmit infra, generating duplicate code and blocks the new direct resources. 

We want to  remove it from the presubmit infra immediately to unblock new resources. But we cannot delete the existing auto-generate code because OSS users may still need it.

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
